framework/PostProcessors/BasicStatistics.py

```diff
@@ -143,6 +143,39 @@ def inputToInternal(self, currentInp):
     if len(currentInput) == 0:
       self.raiseAnError(IOError, "In post-processor " +self.name+" the input "+currentInput.name+" is empty.")
 
+    pbWeights = None
+    if type(currentInput).__name__ == 'tuple':
+      return currentInput
+    # TODO: convert dict to dataset, I think this will be removed when DataSet is used by other entities that
+    # are currently using this Basic Statisitics PostProcessor.
+    if type(currentInput).__name__ == 'dict':
+      if 'targets' not in currentInput.keys():
+        self.raiseAnError(IOError, 'Did not find targets in the input dictionary')
+      inputDataset = xr.Dataset()
+      for var, val in currentInput['targets'].items():
+        inputDataset[var] = val
+      if 'metadata' in currentInput.keys():
+        metadata = currentInput['metadata']
```

Comment:
I thought that the ProbabilityWeight are not in the metadata anymore

Reply:
This is for dictionary. In some other post-processor, dictionary is used as input to basic statistics postprocessor, and ```metadata``` are used to store probabilityweight. When all postprocessor are converted to use DataSet directly, this can be remove.